microsoft-edge/devtools-guide-chromium/css/css-in-js.md

```diff
@@ -0,0 +1,112 @@
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       https://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License. -->
+
+# Style editing for CSS-in-JS frameworks
+
+<!-- from https://docs.microsoft.com/en-us/microsoft-edge/devtools-guide-chromium/whats-new/2020/06/devtools#style-editing-for-css-in-js-frameworks -->
+
+The **Styles** pane supports editing styles that were created with the [CSS Object Model (CSSOM)][CsswgDraftsCssom] APIs.  Many CSS-in-JS frameworks and libraries use the CSSOM APIs under the hood to construct styles.
+
+You can edit styles added in JavaScript using [Constructable Stylesheets][WicgConstructStylesheet].  Constructable Stylesheets are a way to create and distribute reusable styles when using [Shadow DOM][MdnShadowDom].
+
+For example, the `h1` styles added with `CSSStyleSheet` \(CSSOM APIs\) are editable.  The styles are editable in the **Styles** panel.
+
+:::image type="complex" source="../media/css-in-js.msft.png" alt-text="Changing the background property of the h1 styles added with CSSStyleSheet from pink to lightblue" lightbox="../media/css-in-js.msft.png":::
+   Changing the `background` property of the `h1` styles added with `CSSStyleSheet` from `pink` to `lightblue`.
+:::image-end:::
+
+Give this feature a try with a [sample that uses CSS-in-JS][CodepenZoherghadyaliAbdgrpz].
+
+
+## Blog post
+
+This article talks about CSS-in-JS support in DevTools and, in general, what we mean by _CSS-in-JS_ and how it's different from regular CSS.
+
+## What is CSS-in-JS?
+
+The definition of CSS-in-JS is somewhat vague. In a broad sense, it's an approach for managing CSS code using JavaScript. For example, it could mean that the CSS content is defined using JavaScript and the final CSS output is generated on-the-fly by the app.
+
+In the context of DevTools, _CSS-in-JS_ means that the CSS content is injected into the page using CSSOM APIs.  Regular CSS is injected using `<style>` or `<link>` elements, and it has a static source (such as a DOM node or a network resource).  In contrast, CSS-in-JS often doesn't have a static source.  A special case here is that the content of a `<style>` element can be updated using CSSOM API, causing the source to become out of sync with the actual CSS stylesheet.
+
+If you use any CSS-in-JS library (such as styled-component, Emotion, or JSS), the library might inject styles using CSSOM APIs under the hood depending on the mode of development and the browser.
+
+Let's look at some examples on how you can inject a stylesheet using CSSOM API similar to what CSS-in-JS libraries are doing.
```

Comment:
```suggestion
Let's look at some examples on how you can inject a stylesheet using CSSOM API similar to how CSS-in-JS libraries work.
```

Reply:
Ending a sentence with "doing" just sounds weird ... it could just end with "similar to CSS-in-JS libraries." or "in the same manner as CSS-in-JS libraries."  I just made a suggestion to start the conversation.